app/economy/management/commands/fix_earnings.py

```diff
@@ -0,0 +1,157 @@
+        try:
+            tip = tips.get(pk=earning.source_id)
+            if tip.amount:
+                earning.token_value = tip.amount
+                earning.save()
+        except:
+            print(f'unable to find Tip {earning.source_id}')
+
+
+def fixBountyFulfillmentAmount ():
+    print("===========================================")
+    bounty_fulfillments = BountyFulfillment.objects.filter(payout_amount__gt=1000000)
+
+    print(f"Fixing Amount for {bounty_fulfillments.count()}")
+
+    tokens = Token.objects.filter(approved=True)
+
+    for bounty_fulfillment in bounty_fulfillments:
+        token_name = bounty_fulfillment.token_name
+        token = tokens.filter(symbol=token_name).first()
+        if token:       
+            bounty_fulfillment.payout_amount = bounty_fulfillment.payout_amount / (10**token.decimals)
+            bounty_fulfillment.save()
+        else:
+            print(f'error: unable to find {token_name}')
+
+
+class Command(BaseCommand):
+    """Define the management command to fix up earnings."""
+
+    def handle(self, *args, **options):
+        """Fix earnings."""
+
+        print("Pulling in every Earning")
+        # e = Earning.objects.all()
+        # print(f"Total Count: {e.count()}")
+
+        # fetch successful earnings
+        earnings = Earning.objects.all()
+        print(f"Total Count: {earnings.count()}")
+
+        # filter mainnet 
+        earnings_mainnet = earnings.filter(network='mainnet')
+        print(f"Total Mainnet Earnings: {earnings_mainnet.count()}")
+
+
+        # filter earning which have txid
+        earnings_with_txid = earnings_mainnet.filter(txid__isnull=False).exclude(txid__exact='').exclude(txid__exact='0x0')
+        print(f"Total Mainnet Earnings with txnid: {earnings_with_txid.count()}")
+        
+        #  ============================= STEP 1 =======================
+        # fetch only grant contributions
+        earning_contributions = earnings_with_txid.filter(source_type='81')
+        # Ensure Gitcoin contributions are ignored and fixes decimal issue
+        updateContributionEarnings(earning_contributions)
+
+
+        #  ============================= STEP 2 =======================
+        # fetch only kudos and fixes decimal issue
+        earning_kudos = earnings_with_txid.filter(source_type='72')
+        updateKudosEarnings(earning_kudos)
+
+        # Refer below metabse query on why this function needs to be run
+        # https://metabase.gitcoin.co/question/1054
+        fixBountyFulfillmentAmount()
+
+        #  ============================= STEP 3 =======================
+        # fetch only bounty fulfillments and fixes decimal issue
+        earning_bounty_fulfillments = earnings_with_txid.filter(source_type='35')
+        updateBountyEarnings(earning_bounty_fulfillments)
+
+
+        #  ============================= STEP 4 =======================
+        # fetch only bounty fulfillments and fixes decimal issue
```

Comment:
Should say `only tips` instead of `only bounty fulfillments`

Reply:
Will purge this a day after we release. So izz okay
This was simply to get the team to 👀  on what I was trying to do here